packages/ember-core/src/components/scaffold.gts

```diff
@@ -113,6 +115,7 @@ export default class Scaffold extends Component<ScaffoldSignature> {
 
   toggleSidebar = () => {
     this.showSidebar = !this.showSidebar;
+    this.args.onSidebarToggle?.();
```

Comment:
Also, it might be good to include whether or not the sidebar is open, since users have no programmatic way to tell.